### PR TITLE
Feat(eos_cli_config_gen): Add custom comments at the top of the rendered configuration.

### LINKED ISSUE #2271: Add custom comments at the top of the rendered configuration
### Enhancement summary

Opening issue to discuss the option to include custom comments at the top of the generated config. Example:

```
!RANCID-CONTENT-TYPE: arista
!
! Config version: x.x.x
!
daemon attr
...
```

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

It is common to use the rendered config from `eos_cli_config_gen` as tje design configlet in CVP. Sometimes we want to include comments that add information about the configlet such as a generated version or a change ticket, which helps Ops to trace back to the source of the change. This comment section can also be used to force CVP to create a task and reapply a configlet (with no deltas) if we need to do that.

### Describe the solution you would like

a new `header_comments` knob or similar that would allow to add comments at the beginning of the rendered config.

Example:

```yaml
header_comments:
  - "Config version: XXX"
  - "Change ticket: XXXXX"
  - |
    some extra long text
   in multiple lines
```

Would render:

```
!RANCID-CONTENT-TYPE: arista
!
! Config version: XXX
! Change ticket: XXXXX
! some extra long text
! in multiple lines
!
daemon attr
...
```

### Describe alternatives you have considered

Render this information with `eos_cli` or with a custom template, however it gets rendered at the end, not too intuitive if you have to scroll all the way down to see that info

### Additional context

Although comments don't get pushed to the devices, comments in designed configlets for different config section can give some good context about some parts of the config, for those who look at the configlets in CVP.

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Add custom comments at the top of the rendered configuration

## Related Issue(s)

Fixes #2271 

## Component(s) name

`arista.avd.eos_cli_config_gen`

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by MaheshGSLAB — APPROVED** [review]
LGTM, tried on local system with different ways of comment and works fine.

**Review by Shivani-gslab — APPROVED** [review]
LGTM!

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/config-comment.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# config-comment
```
Comment (sugetha24) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/config-comment.md:1]: Confused about this line. Are we expecting the comments to be displayed here?

**Review by sugetha24 — COMMENTED** [review]
Confused about this line. Are we expecting the comments to be displayed here? @Vibhu-gslab
Otherwise, LGTM

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/config-comment.j2 [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#}
+{# eos - config comment #}
+{% if config_comment is arista.avd.defined %}
+!
+{%     set multiline_comment = config_comment.split('\n') %}
+{%     for comment in multiline_comment %}
+{# To avoid empty sting #}
+{%         if comment %}
+!{{ comment }}
+{%         endif %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/config-comment.j2]: I don't think we need to check for empty lines. If someone puts in an empty line, we should print it.
```suggestion
!{{ comment }}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/config_comment.schema.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  config_comment:
+    type: str
+    description: Add custom comment to provide configlet information.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/config_comment.schema.yml]:
```suggestion
    description: |-
      Add a comment to provide information about the configuration.
      This comment will be rendered at the top of the generated configuration.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/config-comment.md
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/config-comment.md:1]: This file must be referred somewhere in `ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md`.
I suggest adding it in the `## Miscellaneous` section.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/config-comment.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+config_comment: |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/config-comment.yml]: Your check for blank lines hides the fact that you are inserting a closing newline with the current input. By adding `-` you avoid that.
```suggestion
config_comment: |-
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/config-comment.j2 (hunk 2 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/config-comment.j2]: Fixed!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/config_comment.schema.yml (hunk 3 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/config_comment.schema.yml]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/config-comment.md
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/config-comment.md:1]: Done!

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/config-comment.yml (hunk 4 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/config-comment.yml]: Done!

**Comment by Vibhu-gslab** [comment]
> Confused about this line. Are we expecting the comments to be displayed here? @Vibhu-gslab Otherwise, LGTM

Could you please specify the line number.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/config-comment.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (sugetha24) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/config-comment.md:2]: @Vibhu-gslab My bad, i forgot to mention this line! Are we expecting the comment to show up here?

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/config-comment.md (hunk 5 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/config-comment.md:2]: @sugetha24, yes this is expected (Its the heading for doc)

**Pull request merged**